### PR TITLE
DOCS-3296 Adds the Synthetics CircleCI Orb README into Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Adds single sourcing logic for CircleCI Orb documentation. 

### Motivation

DOCS-3296

### Preview

https://docs-staging.datadoghq.com/alai97/single-sourcing-ci-orb-docs/synthetics/cicd_integrations/circleci_orb/

### Additional Notes

Remember to add single sourcing logic to the `pull_config.yaml` file before merging! 

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by peculiarism — APPROVED** [review]

File: content/en/synthetics/cicd_integrations/_index.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ To get started, see [Integrations](#integrations) and [use the API](#use-the-api
 ## Integrations
 
 {{< whatsnext desc="With Synthetics and CI/CD, you can run Synthetic tests in the CI platform provider of your choice:" >}}
+    {{< nextlink href="synthetics/cicd_integrations/circleci_orbs" >}}CircleCI Orbs{{< /nextlink >}}
```
Comment (peculiarism) [inline content/en/synthetics/cicd_integrations/_index.md:43]:
```suggestion
    {{< nextlink href="synthetics/cicd_integrations/circleci_orbs" >}}CircleCI Orb{{< /nextlink >}}
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 2]
```diff
@@ -288,6 +288,20 @@
...
+    - action: pull-and-push-file
+      branch: main
+      globs:
+      - 'README.md'
+      options:
+        dest_path: '/synthetics/cicd_integrations/'
+        file_name: 'circleci_orbs.md'
+        front_matters:
+          dependencies: ["https://github.com/DataDog/synthetics-ci-orb/blob/main/README.md"]
+          title: Synthetics CI CircleCI Orbs
```
Comment (peculiarism) [inline local/bin/py/build/configurations/pull_config_preview.yaml:302]:
```suggestion
          title: Synthetics CI CircleCI Orb
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 3]
```diff
@@ -288,6 +288,20 @@
...
 
+  - repo_name: synthetics-ci-orb
+    contents:
+    - action: pull-and-push-file
+      branch: main
+      globs:
+      - 'README.md'
+      options:
+        dest_path: '/synthetics/cicd_integrations/'
+        file_name: 'circleci_orbs.md'
```
Comment (peculiarism) [inline local/bin/py/build/configurations/pull_config_preview.yaml:299]: Can we change the md file to `orb`?

**Review by peculiarism — CHANGES_REQUESTED** [review]
Just a couple of changes but looks good to go!

**Review by apigirl — APPROVED** [review]
This LGTM pending Beths suggestions. Let me know on slack if you need another review/approval

File: config/_default/menus/menus.en.yaml [hunk 4]
```diff
@@ -2309,16 +2309,21 @@ main:
     parent: synthetics_cicd_integrations
     identifier: synthetics_cicd_integrations_configuration
     weight: 501
+  - name: CircleCI Orbs
+    url: /synthetics/cicd_integrations/circleci_orbs
+    parent: synthetics_cicd_integrations
+    identifier: synthetics_cicd_integrations_circleci_orbs
```
Comment (alai97) [inline config/_default/menus/menus.en.yaml:2315]:
```suggestion
  - name: CircleCI Orb
    url: /synthetics/cicd_integrations/circleci_orb
    parent: synthetics_cicd_integrations
    identifier: synthetics_cicd_integrations_circleci_orb
```

File: Makefile [hunk 5]
```diff
@@ -115,6 +115,8 @@ clean-auto-doc: ##Remove all doc automatically created
 	rm -f content/en/serverless/libraries_integrations/macro.md ;fi
 	@if [ content/en/serverless/libraries_integrations/cli.md ]; then \
 	rm -f content/en/serverless/libraries_integrations/cli.md ;fi
+	@if [ content/en/synthetics/cicd_integrations/circleci_orbs.md ]; then \
+	rm -f content/en/synthetics/cicd_integrations/circleci_orbs.md ;fi
```
Comment (alai97) [inline Makefile:119]:
```suggestion
	@if [ content/en/synthetics/cicd_integrations/circleci_orb.md ]; then \
	rm -f content/en/synthetics/cicd_integrations/circleci_orb.md ;fi
```

File: content/en/synthetics/cicd_integrations/_index.md [hunk 6]
```diff
@@ -40,6 +40,7 @@ To get started, see [Integrations](#integrations) and [use the API](#use-the-api
 ## Integrations
 
 {{< whatsnext desc="With Synthetics and CI/CD, you can run Synthetic tests in the CI platform provider of your choice:" >}}
+    {{< nextlink href="synthetics/cicd_integrations/circleci_orbs" >}}CircleCI Orb{{< /nextlink >}}
```
Comment (peculiarism) [inline content/en/synthetics/cicd_integrations/_index.md:43]: Do we need to change the href?

Comment (alai97) [reply content/en/synthetics/cicd_integrations/_index.md:43]:
```suggestion
    {{< nextlink href="synthetics/cicd_integrations/circleci_orb" >}}CircleCI Orb{{< /nextlink >}}
```

**Review by peculiarism — APPROVED** [review]

**Pull request merged**